### ISSUE TITLE
URL encoding of '#' in create_action_hash() causes 404 errors.

### ISSUE THREAD

**Issue by aziz0000** [elementor/elementor#11268]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

The function `create_action_hash` (in `elementor/includes/frontend.php`) is used to create action URLs (e.g. popups, lightboxes, etc.) that start with `#`. For example, a heading element with a link set as `Dynamic Tag`->`Lightbox` produces a URL that looks like this: 

    https://domain/page/#elementor-action:action=lightbox&settings=...

However, the function does URL-encoding on that action part, resulting in a URL that looks like this:

    https://domain/page/%23elementor-action%3Aaction%3Dlightbox%26settings%3D...

Now, this works fine if you just click the link, as it shows the lightbox correctly with no issues. The main problem is that opening it in a new tab results in a 404 page.

The reason is that the `#` is URL encoded as `%23`. Using `%23` in the URL is not the same as `#`. Browsers identify `#` as a fragment identifier, and do not send it as part of the HTTP request ([RFC 3986](https://tools.ietf.org/html/rfc3986)). However, `%23` is not treated the same way. Hence, `github.com/#` works but `github.com/%23` causes 404.

## Steps to reproduce

(This is using Dynamic Tags Lightbox from Elementor Pro)
1. Add new Heading element and set its text to anything
2. Under "Link" click "Dynamic Tags" and select "Action:Lightbox"
3. Click "Lightbox" to show its settings and select an image.
4. Update and view page.
5. Click the link. It should and will show the Lighbox correctly.
6. Right-click the link and select "Open in new tab", this will show a 404 page.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Fix

I was able to fix this by replacing this (in `elementor/includes/frontend.php:1123`):

    return rawurlencode( sprintf( '#elementor-action:action=%1$s&settings=%2$s', $action, base64_encode( wp_json_encode( $settings ) ) ) );

with this:

    return '#' . rawurlencode( sprintf( 'elementor-action:action=%1$s&settings=%2$s', $action, base64_encode( wp_json_encode( $settings ) ) ) );

However, I am not fully familiar with the codebase and I am not sure if this will result in issues with other parts of the code that use the function `create_action_hash`.

## Environment

This issue is system-independent. I have been able to reproduce it on Windows and Linux environments (Apache and Nginx), under different versions of PHP, and on both `localhost` and remote servers.
 
<details>
<summary>System Info</summary>

```
== Active Plugins ==
	Elementor
		Version: 2.9.8

	Elementor Pro
		Version: 2.9.3
```

</details>

**Comment by arielk** [comment]
Duplicate of #11104

**Comment by giyandeva** [comment]
Hello aziz0000,
I tried the mentioned fix, but in this way the pop-up does not work....

**Comment by aziz0000** [comment]
@giyandeva You are right. It seems that it only works for Lightboxes, but not popups.

I looked into it, the issue is related to the binding of the "popup open" action to the element click event. I found a fix for it, but **I must say that I do not know if it is going to break anything else**. So, here it is:

In the file `assets\js\frontend.min.js`, find this string:

    links:'a[href^="%23elementor-action"]'

and replace with

    links:'a[href^="#elementor-action"]'
